index.bs

```diff
@@ -2953,6 +2953,287 @@ value and terminate the operation.
+    [=effective domain=] is not a [=valid domain=], then return [=a promise
+    rejected with=] "{{SecurityError}}" {{DOMException}}.
+1. If |rpId| [=is a registrable domain suffix of or is equal to=]
+    |effectiveDomain|, return [=a promise resolved with=] undefined.
+1. If the client does not support [[#sctn-related-origins|related origin
+    requests]], return [=a promise rejected with=] a "{{SecurityError}}"
+    {{DOMException}}.
+1. Let |p| be [=a new promise=].
+1. Execute the following steps [=in parallel=]:
+    1. If the result of running the [$related origins validation procedure$]
+        with arguments |callerOrigin| and |rpId| is [TRUE], then [=resolve=]
+        |p|.
+    1. Otherwise, [=reject=] |p| with a "{{SecurityError}}" {{DOMException}}.
+1. Return |p|.
+
+#### {{PublicKeyCredential/signalUnknownCredentialId(options)}} #### {#sctn-signalUnknownCredentialId}
+
+Signals that a [=credential id=] was not recognized by the [=[WRP]=], e.g.
+because it was deleted by the user. Unlike
+{{PublicKeyCredential/signalAllAcceptedCredentialIds(options)}}, this
+method does not require passing the entire list of accepted [=credential IDs=]
+and the [=userHandle=], avoiding a privacy leak to an unauthenticated caller
+(see [[#sctn-credential-id-privacy-leak]]).
+
+Upon execution of {{PublicKeyCredential/signalUnknownCredentialId(options)}},
+the [=client=] executes these steps:
+
+1. If the result of [=base64url encoding | base64url decoding=]
+    <code>|options|.{{UnknownCredentialIdOptions/credentialId}}</code> is an
+    error, then return [=a promise rejected with=] a {{TypeError}}.
+1. Let |p| be the result of executing the [$Asynchronous RP ID validation
+    algorithm$] with <code>|options|.{{UnknownCredentialIdOptions/rpId}}</code>.
+1. [=Upon fulfillment=] of |p|, run the following steps [=in parallel=]:
+    1. For every [=authenticator=] presently available on this [=client
+        platform=], invoke the [=signal method/authenticator
+        action/unknownCredentialId=] [=authenticator action=] with |options| as
+        input.
+1. Return |p|.
+
+The <dfn for="signal method/authenticator action">unknownCredentialId</dfn>
+[=signal method/authenticator action=] takes an {{UnknownCredentialIdOptions}}
+|options| and is as follows:
+1. [=map/For each=] [=public key credential source=] |credential| in the
+    [=authenticator=]'s [=credential map=]:
+    1. If the |credential|'s [=public key credential source/rpId=] equals
+        <code>|options|.{{UnknownCredentialIdOptions/rpId}}</code> and the
+        |credential|'s [=public key credential source/id=] equals the result of
+        [=base64url encoding | base64url decoding=]
+        <code>|options|.{{UnknownCredentialIdOptions/credentialId}}</code>,
+        [=map/remove=] |credential| from the [=credentials map=] or employ an
+        [=authenticator=]-specific procedure to hide it from future
+        [=authentication ceremonies=].
```

Comment:
The [explainer mentions](https://github.com/w3c/webauthn/wiki/Explainer:-WebAuthn-Signal-API-explainer#examples) that there are times when one signal might be preferable over another:

> If the user revokes or deletes a credential, e.g. in an account settings UI on the relying party's website, the relying party can opportunistically report this at that time with `signalUnknownCredentialId`. However this will only have effect if the user agent is able to route the report to the same credential provider that created this credential. It may be better to send a `signalAllAcceptedCredentialIds` report instead, with a complete list of valid credential IDs.

However after reviewing the two authenticator actions for these signals, it's unclear to me if the preference for one method over the other (in the use case of a user deleting a credential on the RP's side of things) is one that is reflected in this PR.

Is the main benefit of going with `signalAllAcceptedCredentialIds` after RP-side credential deletion to unhide the credential for some reason later? `signalUnknownCredentialId` doesn't have an affordance for reversing this signal like how the `signalAllAcceptedCredentialIds` authenticator action allows for. Otherwise I can't quite tell how `signalAllAcceptedCredentialIds` might be better for routing the signal to authenticators that may or may not be available when the user deleted the credential.

Reply:
> Otherwise I can't quite tell how signalAllAcceptedCredentialIds might be better for routing the signal to authenticators that may or may not be available when the user deleted the credential.

The credential you just deleted might not be in any reachable provider at that time, but a credential you delete perhaps yesterday might be. SignalAllAcceptedCredentialIds would propagate that information. 

I think my recommendation to an RP would be to always use signalAll... if the user is signed in and so it is safe to do so, and only use signalInvalid... when it isn't.

Reply:
I added a note stating that relying parties should use `signalAllAcceptableCredentials` instead of `signalUnknownCredential` when the user is authenticated to reflect changes that may not have been picked up by the authenticators that are currently attached.

I think there's enough guidance on both methods for RPs now.